File: Benchmarks/ejercicio4/go/main.go
package main

import (
	"encoding/json"
	"fmt"
	"time"
)

func main() {
	start := time.Now()

	data := make([]map[string]interface{}, 2_000_000)
	for i := 0; i < len(data); i++ {
		data[i] = map[string]interface{}{"x": i, "y": fmt.Sprint(i)}
	}
	b, _ := json.Marshal(data)
	var out interface{}
	json.Unmarshal(b, &out)

	elapsed := time.Since(start).Seconds()
	fmt.Printf("Time: %.4f seconds\n", elapsed)
	fmt.Printf("Items processed: %d\n", len(data))
	fmt.Printf("JSON size: %d bytes\n", len(b))
}

File: Benchmarks/ejercicio4/java/main.java
import java.util.ArrayList;
import java.util.List;

class DataItem {
    int x;
    String y;
    
    DataItem(int x, String y) {
        this.x = x;
        this.y = y;
    }
}

public class main {
    public static void main(String[] args) {
        long startTime = System.nanoTime();

        List<DataItem> data = new ArrayList<>();
        for (int i = 0; i < 2_000_000; i++)
            data.add(new DataItem(i, String.valueOf(i)));

        StringBuilder sb = new StringBuilder();
        sb.append("[");
        for (int i = 0; i < data.size(); i++) {
            DataItem item = data.get(i);
            if (i > 0) sb.append(",");
            sb.append("{\"x\":").append(item.x).append(",\"y\":\"").append(item.y).append("\"}");
        }
        sb.append("]");
        String s = sb.toString();
        
        // Simple deserialization simulation (just parse to verify)
        int count = 0;
        for (char c : s.toCharArray()) {
            if (c == '{') count++;
        }

        long endTime = System.nanoTime();
        double elapsed = (endTime - startTime) / 1_000_000_000.0;
        System.out.printf("Time: %.4f seconds%n", elapsed);
        System.out.println("Items processed: " + data.size());
        System.out.println("JSON size: " + s.length() + " bytes");
    }
}
